tools/get_release_contributors.py

```diff
@@ -0,0 +1,134 @@
+
+    def get_login(self, key, commit_hash):
+        if key not in self.keys_to_user:
+            self.keys_to_user[key] = self.get_user_by_commit(commit_hash)
+            self.save()
+        return self.keys_to_user[key]
+
+    def get_user_by_commit(self, hash):
+        headers = {
+            'Accept': 'application/vnd.github.v3+json',
+        }
+        if self.token:
+            headers['Authorization'] = 'token %s' % self.token
+        r = requests.get('https://api.github.com/repos/jamulussoftware/%s/commits/%s' % (self.repo, hash), headers=headers)
+        if 200 <= r.status_code < 300:
+            try:
+                return r.json()['author']['login']
+            except TypeError:
+                logger.warning('%s has not github author, saving as empty', hash)
+                return ''
+        if r.status_code == 422:
+            logger.warning('unable to find author of %s, saving as empty', hash)
+            return ''
+        raise UnexpectedGithubStatus('status was %d' % r.status_code)
+
+    def save(self):
+        with open(self.path, 'w') as f:
+            yaml.dump(self.keys_to_user, f)
+
+    def load(self):
+        with open(self.path, 'r') as f:
+            self.keys_to_user = yaml.safe_load(f) or {}
+
+
+authors_cache = __file__.replace('.py', '-cache.yaml')
+authors = Authors(authors_cache)
+
+
+def main(from_, to):
+    if os.path.exists('src'):
+        logger.info('scanning code repo')
+        authors.set_repo('jamulus')
+        print_code_contributors(from_, to)
+        print_app_translators(from_, to)
+    elif os.path.exists('wiki'):
+        authors.set_repo('jamuluswebsite')
+        print_website_contributors(from_, to)
+        logger.info('scanning website repo')
+    else:
+        logger.error('is this a code or website repo?')
+
+
+def print_code_contributors(from_, to):
+    return print_contributors('Code contributors', ['.', ':!src/res/translation'], from_, to)
+
+
+def print_app_translators(from_, to):
+    return print_contributors('Application translators', ['src/res/translation'], from_, to)
+
+
+def print_website_contributors(from_, to):
+    return print_contributors('Website contributors/translators', ['.'], from_, to)
+
+
+def print_contributors(title, git_log_selector, from_, to):
+    contributors = ['@%s' % u for u in find_contributors(git_log_selector, from_, to) if u and u not in ignore_list]
+    contributors_str = ' '.join(contributors)
+    print('%s: %s' % (title, contributors_str))
+
+
+def find_contributors(git_log_selector, from_, to):
+    contributors = set()
+    commits = subprocess.check_output(['git', 'log', '--format=format:%H %an <%ae>', '%s..%s' % (from_, to), '--'] + git_log_selector)
+    commits = commits.decode('utf-8')
+    for commit in commits.split('\n'):
+        hash, author_key = commit.split(' ', 1)
+        login = authors.get_login(author_key, hash)
+        contributors.add(login)
+    return sorted(contributors)
```

Comment:
I'm not very familiar with python, but have tried running this script. Is it possible to sort the list of usernames case-insensitively? Currently, `@SeeLook` and `@Snayler` come before `@ann0see`.

Reply:
Done, will update the announcement as well.